packages/utilities/src/lib/getDeepObjectKeys.ts

```diff
@@ -1,47 +1,61 @@
-import { isObject } from './isObject';
+import { isNullOrUndefinedOrEmpty } from './isNullOrUndefinedOrEmpty';
 import type { AnyObject } from './utilityTypes';
 
 /**
  * Flattens an object to a list of its keys, traversing deeply into nested objects and arrays of objects.
  *
- * @note By default Nested array values are flattened to `arrayKey.${index}.subKey`. This can be changed to `arrayKey[${index}].subKey` by setting `arrayKeysBracedIndex` to `true`.
+ * @note By default Nested array values are flattened to `arrayKey.${index}.subKey`.
+ * This can be changed to `arrayKey[${index}].subKey` by setting `options.arrayKeysIndexStyle` to `'braces-with-dot'`.
+ * Or it can also be changed to `arrayKey[${index}]subKey` by setting `options.arrayKeysIndexStyle` to `'braces'`.
  *
  * @param obj The object of which to deeply retrieve its keys
+ * @param options The options with which to customize the output of this function
  * @returns An array of strings holding the keys of the object
  */
-export function getDeepObjectKeys<T>(
+export function getDeepObjectKeys<T>(obj: AnyObject<T>, options?: GetDeepObjectKeysOptions): string[] {
+	return [...getDeepObjectKeysGenerator(obj, options)];
+}
+
+function* getDeepObjectKeysGenerator<T>(
 	obj: AnyObject<T>,
 	{ arrayKeysIndexStyle = 'dotted' }: GetDeepObjectKeysOptions = { arrayKeysIndexStyle: 'dotted' }
-): string[] {
-	const keys: string[] = [];
+): Generator<string> {
+	if (Array.isArray(obj)) {
+		for (const [index, value] of obj.entries()) {
+			const resolvedIndex = arrayKeysIndexStyle === 'dotted' ? `${index}` : arrayKeysIndexStyle === 'braces' ? `[${index}]` : `[${index}].`;
```

Comment:
Can you please turn this into its own method just for clarity sake? It won't slow down much anyways but it'll help with reading the code (and not missing a fix in half of it)